src/pages/index.js

```diff
@@ -17,131 +21,185 @@ import ModalWithForm from "../components/ModalWithForm.js";
+            })
+            .catch((err) =>
+              console.log(`An error occurred when adding a like: ${err}`)
+            );
+        }
+      },
+      cardSelector: selectors.cardTemplate
+    },
+    
+  );
+  return card.getView();
+};
 
 // Instance of the UserInfo class
-const userInfo = new UserInfo({
-  userNameSelector: ".profile__name",
-  userDescriptionSelector: ".profile__description",
-  userImageSelector: ".profile__image",
+const newUserInfo = new UserInfo({
+  nameSelector: "#profileName",
+  descSelector: "#profileDescription",
+  avatarSelector: "#profileImage",
 });
 
-// Create and enable form validators for editing profile and adding cards
-const editProfileValidator = new FormValidator(config, profileForm);
-editProfileValidator.enableValidation();
-const editCardValidator = new FormValidator(config, cardAddModal);
-editCardValidator.enableValidation();
+// get web server info, cards and user data.
+api
+  .getWebpageInfo()
+  .then(([cardData, userData]) => {
+    userId = userData._id;
+    cardSection = new Section(
+      {
+        items: cardData,
+        renderer: (data) => {
+          const cardElement = card(data);
+          cardSection.addItem(cardElement);
+        },
+      },
+      ".cards"
+    );
+    cardSection.renderItems();
+    newUserInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+    });
+    newUserInfo.setAvatar(userData.avatar);
+  })
+  .catch((err) =>
+    console.log(
+      `An error occurred when loading initial user and card data: ${err}`
+    )
+  );
+
+const cardAddForm = new ModalWithForm("#cardAddModal", (data) => {
+  const newCard = { name: data['card-title-input'], link: data['card-image-input'] };
+  cardAddForm.setApiCalling(true);
+  api
+    .addCard(newCard)
+    .then((result) => {
+      const cardElement = card(result);
+      cardSection.addItem(cardElement);
+      cardAddForm.close();
+    })
+    .catch((err) =>
+      console.log(`An error occurred when loading new card data: ${err}`)
+    )
+    .finally(() => cardAddForm.setApiCalling(false));
+});
+cardAddForm.setEventListeners();
 
-// Create an instance of the Section class
-const section = new Section(
-  { items: initialCards, renderer: (item) => renderCard(item, section) },
-  ".cards"
+addCardButton.addEventListener("click", () => {
+  addFormValidator._toggleButtonState();
+  cardAddForm.open();
+});
+const addFormValidator = new FormValidator(
+  validationConfig,
+  selectors.addForm
 );
+addFormValidator.enableValidation();
+
+const editProfileForm = new ModalWithForm("#profileEditModal", (data) => {
+  editProfileForm.setApiCalling(true);
+  api.getProfileInfo(data)
+    .then(() => {
+      newUserInfo.setUserInfo(data);
+      editProfileForm.close();
+    })
+    .catch((err) =>
+      console.log(`An error occurred when loading user profile data: ${err}`)
+    )
+    .finally(() => editProfileForm.setApiCalling(false));
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```